fontbe/src/features/marks.rs

```diff
@@ -305,6 +334,63 @@ impl<'a> MarkLookupBuilder<'a> {
         }
         result
     }
+
+    fn make_mark_to_liga_groups(&self) -> BTreeMap<GroupName, MarkGroup<'_>> {
+        let mut groups = BTreeMap::<_, MarkGroup>::new();
+        let mut liga_anchor_groups = HashSet::new();
+
+        // a temporary buffer, reused for each glyph
+        let mut component_groups = HashMap::new();
+        for (glyph_name, anchors) in &self.anchor_lists {
+            let is_not_liga = !self.gdef_classes.is_empty()
+                && (self.gdef_classes.get(glyph_name) != Some(&GlyphClassDef::Ligature));
+            if is_not_liga {
+                continue;
+            }
+            let Some(max_index) = anchors.iter().filter_map(|a| a.ligature_index()).max() else {
+                // no max index means no actual ligature attachments?
+                continue;
+            };
+            for anchor in anchors {
+                let AnchorKind::Ligature { group_name, index } = &anchor.kind else {
+                    continue;
+                };
+                liga_anchor_groups.insert(group_name);
+                let component_anchors = component_groups
+                    .entry(group_name.clone())
+                    .or_insert_with(|| vec![None; max_index]);
+                component_anchors[*index - 1] = Some(*anchor);
+            }
+
+            for (group_name, anchors) in component_groups.drain() {
+                groups
+                    .entry(group_name)
+                    .or_default()
+                    .bases
+                    .push((glyph_name.clone(), BaseOrLigaAnchors::Liga(anchors)));
+            }
+        }
+
+        // then we do another pass to add the marks:
```

Comment:
It would be nice to have an equivalently clear comment explaining what the first pass does